### ISSUE TITLE
How to listen header prev,next click events?

### ISSUE THREAD

**Issue by yuta17** [fullcalendar/fullcalendar-vue#124]
Any updates on this for v5?
What is the correct way of listening to date changes?

https://github.com/fullcalendar/fullcalendar-vue/issues/24#issuecomment-675607967

**Comment by kgilden** [comment]
@yuta17 you can probably use [`datesSet`](https://fullcalendar.io/docs/datesSet).

**Comment by acerix** [comment]
Please refer to the [support page](https://fullcalendar.io/support/) and use [Stack Overflow](https://stackoverflow.com/questions/tagged/fullcalendar) for help.  If this is a bug, please supply a [runnable, stripped-down demonstration](http://fullcalendar.io/wiki/Reduced-Test-Cases/).

**Comment by yuta17** [comment]
Thanks!

**Comment by DKWLB** [comment]
> @yuta17 you can probably use [`datesSet`](https://fullcalendar.io/docs/datesSet).

I tried, but i get infinite fetch request...

**Comment by DKWLB** [comment]
Sorry, but i will write  here,maybe someone will reply soon.
I've spent whole night but none of works until the end. 
Built in ajax - event functions work fine, but eventClickHandler - brings me not the format i need. For editinng form. Extraprops and so on.
So i make independent initial array of events using calendar api ActiveEnd and Active Start. But i really need to listen prev and next events.
DataSet make infinite requests.

**Comment by safiullahsarhandi** [comment]
> @yuta17 you can probably use [`datesSet`](https://fullcalendar.io/docs/datesSet).

@kgilden your solution works for me

**Comment by guillaumebriday** [comment]
I run into the same issue @DKWLB, did you solve it?

**Comment by DariaYeremina** [comment]
@guillaumebriday trying to avoid infinite loop, I've set `lastStartDate` and `lastEndDate`. And my handler looks like this
`handleDatesSet (event) {
            if (this.lastStartDate !== event.startStr && this.lastEndDate !== event.endStr) {
                this.$emit('set-dates', {
                    start: event.startStr,
                    end: event.endStr
                });
                this.lastStartDate = event.startStr;
                this.lastEndDate = event.endStr;
            }
        }`

**Comment by guillaumebriday** [comment]
thanks @DariaYeremina 

@acerix Can we prevent `datesSet` to be called anytime we add an event in the calendar? 🤔

**Comment by reecehill** [comment]
Yeah, I'm still having problems with this too! Bug?

**Comment by reecehill** [comment]
For those struggling, I've managed to come up with a workaround. It's not nice, but it'll do for now. My issue was that datesSet would lead to a loop, where upon the success of one call another API call was made. 

I've dropped the datesSet hook, and instead created [custom buttons](https://fullcalendar.io/docs/customButtons) for Next, Prev, Today etc.

An example of a button:
`customButtons: {
customNext: {
						text: "Next",
						bootstrapFontAwesome: "right-single-arrow",
						click: () => {
							this.nextButtonHandler();
						},
					},
					}
`

And my nextButtonHandler access the calendar's api and manually triggers the next view, then calls my api with the currently active date range.

I've not tested this at all, but it's 2am and it'll do!

`		nextButtonHandler() {
			this.$refs["full-calendar"].getApi().next();
			this.getEventsInView(
				this.$refs["full-calendar"].getApi().currentData.dateProfile.currentRange
			);
		},`
		

getEventsInView() is my API call (though I use Vuex).

Hope this helps somebody!
